e2eAIOK/ModelAdapter/training/train.py

```diff
@@ -112,11 +112,18 @@ def _dist_wrapper(self):
         """            
         if ext_dist.my_size > 1:
             self.logger.info("training with DistributedDataParallel")
-            self.model = ext_dist.DDP(self.model)
-            self.model.loss = MethodType(lambda obj,*args: obj.module.loss(*args) ,self.model) #  obj stands for model, both original model and transferrable model
-            if self.is_transferrable: # dispatch method to model.module
-                self.model.get_backbone = MethodType(lambda obj: obj.module.get_backbone() ,self.model) # obj stands for model
-                self.model.get_training_metrics = MethodType(lambda obj,*args: obj.module.get_training_metrics(*args) ,self.model) #  obj stands for model
+            if self.is_transferrable:
+                self.model.backbone = ext_dist.DDP(self.model.backbone)
+                self.model.backbone.loss = MethodType(lambda obj,*args: obj.module.loss(*args) ,self.model.backbone) #  obj stands for model, both original model and transferrable model
+                if self.model.distiller is not None and not self.model.distiller.is_frozen:
+                    self.model.distiller = ext_dist.DDP(self.model.distiller) # if distiller require_grad=False, may fail in this step, just skip
+                if self.model.adapter is not None:
+                    self.mode.adapter = ext_dist.DDP(self.mode.adapter)
+            else:
+                self.model = ext_dist.DDP(self.model)
+                self.model.loss = MethodType(lambda obj,*args: obj.module.loss(*args) ,self.model) #  obj stands for model, both original model and transferrable model
+                #     self.model.get_backbone = MethodType(lambda obj: obj.module.get_backbone() ,self.model) # obj stands for model
+                #     self.model.get_training_metrics = MethodType(lambda obj,*args: obj.module.get_training_metrics(*args) ,self.model) #  obj stands for model
```

Comment:
Here you may not comment these two lines, because your model is wrapped by DDP, directly take the attribute will report a error, please test this use case: when you use DDP and your model is not transferrable

Reply:
backbone.loss is applied in task.py. 125 and 126 is not needed when it's not a transferrable mode.